index.coffee

```
@@ -295,20 +298,22 @@ exports.DockerProgress = class DockerProgress
 			callback = options
 			options = null
 
+		ignoreErrorEvents = !!options?.ignoreProgressErrorEvents
```

Here I was faced with the question of what the default value should be, if the caller does not provide the `ignoreProgressErrorEvents` option. Having browsed the code in `resin-builder`, `resin-multibuild` and `balena-cli`, I think we reduce overall code-changing effort by detecting errors by default (although requiring a change type of 'major'). Most importantly, new uses of docker-progress are less likely to accidentally ignore errors.

---

I was wondering about this myself as well, seeing the PR. I think, unless we can think of a good reason, it would be better to just propagate the errors like you do in the PR without the ability to opt out. That means a major version bump.

Otherwise, I think it’s better the default of the flag would preserve the current behaviour and clients could switch it on when they are updated to take advantage of the new behaviour.

---

I'm personally of the view it's better to propagate up and ensure the errors are seen. And I can't think of a reason not to (but yes, major bump).